static/src/javascripts/projects/common/modules/experiments/affix.js

```diff
@@ -0,0 +1,116 @@
+// @flow
+
+import bonzo from 'bonzo';
+import mediator from 'lib/mediator';
+import fastdom from 'lib/fastdom-promise';
+
+const getPixels = (top: string): number =>
+    top !== 'auto' ? parseInt(top, 10) : 0;
+
+class Affix {
+    affixed: boolean;
+    $markerTop: bonzo;
+    $markerBottom: bonzo;
+    $container: bonzo;
+    $element: bonzo;
+    $window: bonzo;
+
+    constructor(options: {
+        element: ?HTMLElement,
+        topMarker: ?HTMLElement,
+        bottomMarker: ?HTMLElement,
+        containerElement: ?HTMLElement,
+    }): void {
+        window.addEventListener('click', () => {
+            this.checkPosition();
+        });
+
+        mediator.addListener('window:throttledScroll', () => {
+            this.checkPosition();
+        });
+
+        mediator.addListener('window:throttledResize', () => {
+            fastdom.write(() => {
+                this.calculateContainerPositioning();
+            });
+        });
+
+        this.affixed = false;
+        this.$markerTop = bonzo(options.topMarker);
+        this.$markerBottom = bonzo(options.bottomMarker);
+        this.$container = bonzo(options.containerElement);
+        this.$element = bonzo(options.element);
+        this.$window = bonzo(document.body);
+
+        this.checkPosition();
+
+        fastdom.write(() => {
+            this.calculateContainerPositioning();
+        });
+    }
+
+    calculateContainerPositioning(): void {
+        // aleady called from inside a fastdom.write cb...
```

Comment:
Could you move the `fastdom.write` inside of the method, so we don't wrap another `read` and `write` into `write`?